src/app/CreateRecording/CreateRecording.tsx

```diff
@@ -0,0 +1,262 @@
+  provider: string;
+}
+
+export const RecordingNamePattern = /^[\w_]+$/;
+export const TemplatePattern = /^template=([\w]+)$/;
+export const EventSpecifierPattern = /([\w\\.\$]+):([\w]+)=([\w\\d\.]+)/;
+
+export const CreateRecording = (props: CreateRecordingProps) => {
+  const context = React.useContext(ServiceContext);
+  const notifications = React.useContext(NotificationsContext);
+  const history = useHistory();
+
+  const [recordingName, setRecordingName] = React.useState(props.recordingName || '');
+  const [nameValid, setNameValid] = React.useState(false);
+  const [continuous, setContinuous] = React.useState(false);
+  const [duration, setDuration] = React.useState(30);
+  const [durationUnit, setDurationUnit] = React.useState(1);
+  const [templates, setTemplates] = React.useState([] as EventTemplate[]);
+  const [template, setTemplate] = React.useState('');
+  const [eventSpecifiers, setEventSpecifiers] = React.useState('');
+  const [eventsValid, setEventsValid] = React.useState(false);
+
+  const handleContinuousChange = (checked, evt) => {
+    setContinuous(evt.target.checked);
+  };
+
+  const handleDurationChange = (evt) => {
+    setDuration(Number(evt));
+  };
+
+  const handleDurationUnitChange = (evt) => {
+    setDurationUnit(Number(evt));
+  };
+
+  const handleTemplateChange = (name) => {
+    setEventsValid(!!name);
+    setEventSpecifiers('');
+    setTemplate(name);
+  };
+
+  const handleEventSpecifiersChange = (evt) => {
+    setEventsValid(TemplatePattern.test(evt) || EventSpecifierPattern.test(evt));
+    setTemplate('');
+    setEventSpecifiers(evt);
+  };
+
+  const getEventSpecifiers = () => !!template ? `template=${template}` : eventSpecifiers;
+
+  const getEventString = () => !!template ? `template=${template}` : eventSpecifiers.split(/\s+/).filter(Boolean).join(',');
+
+  const handleRecordingNameChange = (name) => {
+    setNameValid(RecordingNamePattern.test(name));
+    setRecordingName(name);
+  };
+
+  const handleSubmit = () => {
+    const eventString = getEventString();
+
+    let notificationMessages: string[] = [];
+    if (!nameValid) {
+      notificationMessages.push(`Recording name ${recordingName} is invalid`);
+    }
+    if (!eventsValid) {
+      notificationMessages.push(`Event specifiers are invalid`);
+    }
+    if (notificationMessages.length > 0) {
+      const message = notificationMessages.join('. ').trim() + '.';
+      notifications.warning('Invalid form data', message);
+      return;
+    }
+
+    const command = continuous ? 'start' : 'dump';
+    const id = context.commandChannel.createMessageId();
+    context.commandChannel.onResponse(command).pipe(
+      filter(m => m.id === id),
+      first(),
+      )
+      .subscribe((msg) => {
+          if (msg.status === 0) {
+            notifications.success('Recording created');
+            history.push('/recordings');
+          } else {
+            notifications.danger(`Request failed (Status ${msg.status})`, msg.payload)
```

Comment:
The payload looks like:

Danger alert:Request failed (Status -2)
FlightRecorderException: com.redhat.rhjmc.containerjfr.core.templates.TemplateService$UnknownEventTemplateException: Unknown event template "def"

Can it show a more user-friendly message? I'd recommend "Create recording failed: Unknown event template "def".  Just dropping the stack trace would be fine for a start.

Reply:
I could do something like that, although what you see there is just it dumping what it receives from ContainerJFR into a string. There would need to be some basic parsing or at least regex match to get it to process the message and drop the stack trace, or else ContainerJFR would need to be modified to send a different response.